Add new config for PrettyPrinter to minimize empty tags

ashawley opened the pull request:
This commit containing a unit test was mentioned last year in #46, and I thought the copy constructor using `minimizeEmpty=true` was broken, but it appears scala-xml does the right thing!

debanne commented:
Hello, when the option minimizeEmpty is set to true,  PretttyPrinter still transforming "<a ... />" to "<a>\n    </a>" when the line is longer than "width" or does it prevent this behaviour?

ashawley commented:
@debanne Thanks for the note.  This PR is just for providing a method to use the existing feature to minimize empty tags by adding a class parameter to configure as such.  If there are existing issues with the pretty printer's behavior, they aren't taken up here.

So, the splitting of empty tags, making them open-close tags, and then annoyingly spreading them across two lines: That will need to be a separate issue or PR.

Here's one unit test, to get it started, though!:

```
  @UnitTest
  def prettyPrintAddsNewlines: Unit = {
    val pp = new xml.PrettyPrinter(4, 2)
    val x = <a/>
    // What it does:
    // assertEquals("<a>\n</a>", pp.format(x))
    // What it should do:
    assertEquals("<a/>", pp.format(x))
  }
```

This assumes I've understood your question.

SethTisue commented:
needs rebase now.

other than that, where do we stand, here...?

ashawley commented:
Rebased.

This fixes half of #46 where it doesn't have a minimize empty option.  I've renamed the issue so it doesn't auto-close.  The other issues in #46 will have to be taken up separately.

hosamaly commented on shared/src/main/scala/scala/xml/PrettyPrinter.scala:
```diff
@@ -23,8 +23,11 @@ import Utility.sbToString
  *  @param step  indentation
  */
-class PrettyPrinter(width: Int, step: Int) {
+class PrettyPrinter(width: Int, step: Int, minimizeEmpty: Boolean) {
```
Could you please document the expected behaviour when this parameter is set to true? (e.g. in `@param`)
